Realm/RLMObjectSchema.mm

```diff
@@ -281,14 +281,21 @@ + (NSArray *)propertiesForClass:(Class)objectClass isSwift:(bool)isSwiftClass {
                     // Managed lazy vars are't allowed.
                     // NOTE: Revisit this once property behaviors are implemented in Swift.
                     NSString *lazyPropertyBaseName = [self baseNameForLazySwiftProperty:propertyName];
-                    if (lazyPropertyBaseName && ![ignoredProperties containsObject:lazyPropertyBaseName]) {
-                        @throw RLMException(@"Lazy managed property '%@' is not allowed on a Realm Swift object class. Either add the property to the ignored properties list or make it non-lazy.", lazyPropertyBaseName);
+                    if (lazyPropertyBaseName) {
+                        if ([ignoredProperties containsObject:lazyPropertyBaseName]) {
+                            // This property is the storage property for a ignored lazy Swift property. Just continue.
+                            return;
+                        } else {
+                            @throw RLMException(@"Lazy managed property '%@' is not allowed on a Realm Swift object class. Either add the property to the ignored properties list or make it non-lazy.", lazyPropertyBaseName);
+                        }
+                    } else {
```

Comment:
I personally tend to avoid `else`s after `return`s, but I don't think that we as a team have a guideline one way or the other.